### ISSUE TITLE
SPAWN does not teleport back to location

### ISSUE THREAD

**Issue by NighteranceMC** [lenis0012/LoginSecurity-2#107]
If I choose option SPAWN, it does not teleport player back to last location, just teleports to spawn, so he stays there. There is no error.

**Comment by lenis0012** [comment]
What version are you using @NighteranceMC

**Comment by bruce965** [comment]
Same error for me *LoginSecurity-2.1.7-Spigot.jar*.

**Comment by nelmaster** [comment]
I have the same problems with 2.1.7 SPAWN not work and RANDOM , using bukkit server 1.12

Some help for fix this please !!
Thank

**Comment by bruce965** [comment]
OK, I managed to locate the bug.

I'll be honest though: I don't really understand why it happens as I'm not a Java developer. I'm guessing getters' in `PlayersLocation` are patched at runtime to fetch data from the database, that would explain this behavior.

It's really just a matter of adding `String world = getWorld();` in the method that converts a `PlayerLocation` to the equivalent `Location`.

Here is a patch to fix this bug, to be applied to [4709b9e](https://github.com/lenis0012/LoginSecurity-2/commit/4709b9e26ed7ce27c0ab97eb111e8eb13f84d8e7).

Thanks for your work and your support, this really is the #​1 Auth plugin!

```patch
commit 56d31daf3a81754d86aaa22286a21a2cb5ca3daf
Author: Fabio Iotti <fabiogiopla@gmail.com>
Date:   Sun Jul 2 10:33:54 2017 +0200

    Fix SPAWN location

diff --git a/src/main/java/com/lenis0012/bukkit/loginsecurity/storage/PlayerLocation.java b/src/main/java/com/lenis0012/bukkit/loginsecurity/storage/PlayerLocation.java
index bd6ae44..a427249 100644
--- a/src/main/java/com/lenis0012/bukkit/loginsecurity/storage/PlayerLocation.java
+++ b/src/main/java/com/lenis0012/bukkit/loginsecurity/storage/PlayerLocation.java
@@ -44,6 +44,7 @@ public class PlayerLocation {
     }
 
     public Location asLocation() {
+        String world = getWorld();
         if(world == null) {
             return null;
         }

```

**Comment by lenis0012** [comment]
@bruce965 That is extremely odd indeed.
I am currently in the process of refactoring the database layer, because spigot removed Ebean ORM.
(this is the reason why 2.1.7 has a much larger file size than 2.1.6)

I will look into implementing this into a 2.1.8 hotfix though.

**Comment by nelmaster** [comment]
Please can you share the PlayerLocation.class fixit ?

Thank !!!

**Comment by lenis0012** [comment]
@nelmaster Absolutely, I will do it tonight

**Comment by bruce965** [comment]
Be advised that 2.1.7 patched with my modification is broken when the player disconnects from the end. Not sure about the original.

**Comment by lenis0012** [comment]
@bruce965 I am heading off to bed right now.
If you're willing to, could you try running http://ci.lenis0012.com/job/LoginSecurity/8/

And seeing if it works?

**Comment by bruce965** [comment]
@lenis0012 Sorry, I forgot to reply.

I tried the version you linked and it seems to still have problems with the End World.

If my help is appreciated I might try to see if I can do something about this problem, as soon as I'll be available.

For now: thanks and have a nice weekend.

**Comment by lenis0012** [comment]
Should be fixed as of 3.0.0
